pkg/apis/backupdriver/v1alpha1/snapshot.go

```diff
@@ -32,16 +32,17 @@ type SnapshotSpec struct {
 type SnapshotPhase string
 
 const (
-	SnapshotPhaseNew            SnapshotPhase = "New"
-	SnapshotPhaseInProgress     SnapshotPhase = "InProgress"
-	SnapshotPhaseSnapshotted    SnapshotPhase = "Snapshotted"
-	SnapshotPhaseSnapshotFailed SnapshotPhase = "SnapshotFailed"
-	SnapshotPhaseUploading      SnapshotPhase = "Uploading"
-	SnapshotPhaseUploaded       SnapshotPhase = "Uploaded"
-	SnapshotPhaseUploadFailed   SnapshotPhase = "UploadFailed"
-	SnapshotPhaseCanceling      SnapshotPhase = "Canceling"
-	SnapshotPhaseCanceled       SnapshotPhase = "Canceled"
-	SnapshotPhaseCleanupFailed  SnapshotPhase = "CleanupAfterUploadFailed"
+	SnapshotPhaseNew                    SnapshotPhase = "New"
+	SnapshotPhaseInProgress             SnapshotPhase = "InProgress"
+	SnapshotPhaseSnapshotted            SnapshotPhase = "Snapshotted"
+	SnapshotPhaseSnapshotFailed         SnapshotPhase = "SnapshotFailed"
+	SnapshotPhaseUploading              SnapshotPhase = "Uploading"
+	SnapshotPhaseUploaded               SnapshotPhase = "Uploaded"
+	SnapshotPhaseUploadFailed           SnapshotPhase = "UploadFailed"
+	SnapshotPhaseCanceling              SnapshotPhase = "Canceling"
+	SnapshotPhaseCanceled               SnapshotPhase = "Canceled"
+	SnapshotPhaseCleanupFailed          SnapshotPhase = "CleanupAfterUploadFailed"
+	SnapshotPhaseUploadFailedAfterRetry SnapshotPhase = "UploadFailedAfterRetry"
```

Comment:
Will be good to add comments to explain how SnapshotPhaseUploadFailedAfterRetry is different from SnapshotPhaseUploadFailed.  Why do we need both?

Reply:
My understanding is each upload status  need to have corresponding snapshot status and we set the snapshot status based on upload status in function "syncUploadByKey".

Reply:
Please review the comments I added in the code.  After we agree the wording are fine, I will add them to README.